### ISSUE TITLE
Override the indicator style

### ISSUE THREAD

**Issue by lord007tn** [prevwong/craft.js#122]
I didn't know how I can override the indicator and make a custom one maybe have the shape of the dragged component besides that green or red line.

in Docs indicator can only have success color and error color passed from the editor as a string. maybe the indicator can be a whole HTML element. where you can make your custom one and pass it to the editor state to use it.

**Comment by RaGreen** [comment]
Render indicator? Sadly it's hard coded inside packages/utils/RenderIndicator.tsx

IMHO all indicators should be divs with classes like 'craftjs-render-indicator' to support some customization.

**Comment by oc-Design** [comment]
Can setProp add other attributes to an element?

**Comment by lord007tn** [comment]
> Can setProp add other attributes to an element?

nope, setProp is more of a useState function where you can change the initial state using it.
so when you use setProp you actually setting the value of the declared props of your component. 

but you can add other attributes to the element take a look at this where i add the transformation prop from the element state:
**transformation are inside the useNode() hook**
![image](https://user-images.githubusercontent.com/38087431/92397333-8cb1d900-f11e-11ea-877a-db9e582a8696.png)

I hope this answers your question

**Comment by MarkLyck** [comment]
🙏please make this a function. 

That is probably the biggest improvement craft.js could get, actually rendering the correct space for the component you are dragging as react-beautiful-dnd does it: https://github.com/atlassian/react-beautiful-dnd

I was hoping I could at the very least make it a custom component.

**Comment by TheGrimSilence** [comment]
Personally, without complicating things, I'd like to be able to customize the indicator and transitions. The indicator warping all over the place is unnecessary to me I'd prefer it just snap to the next available drop zone.
